### ISSUE TITLE
%apphelp removes whitespaces/indentation

### ISSUE THREAD

**Issue by guma44** [apptainer/singularity#1391]
## Version of Singularity: 2.4.2-dist

The %apphelp section strips indentation and empty lines.

### Expected behavior

Please keep the indentation and new lines. The help for an app is not readable at all.

### Steps to reproduce behavior

For the recipe:
```
BootStrap: docker
From: ubuntu:16.04

%apprun test
echo "Test app"

%apphelp test
This is a help

   without whitespaces

%help
This is a help

   with whitespaces


```

Running:
```
singularity help test.img
```
gives:
```

This is a help

   with whitespaces

```
And running:

```
singularity help --app test test.img
```

gives:
```
This is a help
without whitespaces
```

**Comment by guma44** [comment]
So the question really is if the lack of whitespaces and newlines is crucial? At least in thI could easily fix this by adding IFS="" to get_section and removing if statemen ie. going from:
```
get_section() {

    SECTION="$1"
    RECIPE="$2"
    FOUND="no"

    while read -r line
    do

 
        if echo $line | grep -q -e "^%$SECTION$"; then

            # If not found, continue
            if [[ "$FOUND" == "no" ]]; then
                FOUND="yes"
            else
                return 0
            fi

        # We didn't match this line but...
        else
            # It's the start of a new section
            if echo $line | grep -q -e "^%"; then

                # We already found it, we're done
                if [[ "$FOUND" == "yes" ]]; then
                    return 0
                fi

            else
                # We already found it, still in section
                if [[ "$FOUND" == "yes" ]]; then
                    if [ ! -z "${line}" ]; then
                        echo $line
                    fi 

                fi            
            fi
        fi

    done < $RECIPE

}
```
To:
```
get_section() {

    IFS=""
    SECTION="$1"
    RECIPE="$2"
    FOUND="no"

    while read -r line
    do
        if echo $line | grep -q -e "^%$SECTION$"; then

            # If not found, continue
            if [[ "$FOUND" == "no" ]]; then
                FOUND="yes"
            else
                return 0
            fi

        # We didn't match this line but...
        else
            # It's the start of a new section
            if echo $line | grep -q -e "^%"; then

                # We already found it, we're done
                if [[ "$FOUND" == "yes" ]]; then
                    return 0
                fi

            else
                # We already found it, still in section
                if [[ "$FOUND" == "yes" ]]; then
                        echo $line
                fi            
            fi
        fi

    done < $RECIPE
}
```

This would mean that just by defining %apphelp section the file would be created. The complete solution would be to rewrite `get_section` function to build the lines and check them for emptiness as a whole.

**Comment by dtrudg** [comment]
Hi @bauerm97 - are you able to comment on the above, and PR #1404 - since this may need consideration for the Golang parsing also? Thanks.

**Comment by dtrudg** [comment]
Closing this ticket as the fix was included in 2.5.0. Thank you @guma44 !

**Comment by guma44** [comment]
Your welcome :)
